R/readLPJmLClimateInput.R

```diff
@@ -0,0 +1,197 @@
+                                         n = 5, endian = .Platform$endian))
+      startyear    <- inHeader[1]
+      nyear        <- inHeader[2]
+      noAnnualPredictions <- inHeader[5]
+      years        <- seq(startyear, startyear + nyear - 1, 1)
+      close(filedata)
+
+    } else {
+      stop("File format of LPJmLClimateInput data unknown. Please provide .clm file format.")
+    }
+
+    if (subset == "wetDaysMonth") {
+
+      if (subtype$variable != "precipitation") stop("Subset 'wetDaysMonth' is only
+                                                   available for 'precipitation'")
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420,
+                                    rule4binary = ">0") / noAnnualPredictions
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset == "annualMean") {
+
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420) / noAnnualPredictions
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset == "annualSum") {
+
+      x <- lpjclass::read.LPJ_input(file_name   = filename,
+                                    out_years   = paste0("y", years),
+                                    namesum     = TRUE,
+                                    ncells      = 67420)
+
+      class(x) <- "array"
+      x        <- collapseNames(as.magpie(x, spatial = 1))
+
+
+    } else if (subset %in% c("monthlyMean", "monthlySum")) {
+      # define year sets (cut it in bunches)
+      bunchLength <- 1
+      yearsets    <- split(years, ceiling(seq_along(years) / bunchLength))
+
+      # define month mapping
+      monthLength <- c(jan = 31, feb = 28, mar = 31, apr = 30,
+                       may = 31, jun = 30, jul = 31, aug = 31,
+                       sep = 30, oct = 31, nov = 30, dec = 31)
+      daysMonth   <- NULL
+      for (m in 1:12) {
+        daysMonth <- c(daysMonth, rep(names(monthLength[m]),
+                                      monthLength[m]))
+      }
+      month2day   <- cbind(day   = 1:sum(monthLength),
+                           month = daysMonth)
+      monthLength <- as.magpie(monthLength)
+
+      # create output object
+      x <- NULL
+
+      # loop over bunches
+      for (b in seq_along(yearsets)) {
+        # read in a bunch of years
+        tmp <- lpjclass::read.LPJ_input(file_name = filename,
+                                        out_years = paste0("y", yearsets[[b]]),
+                                        namesum   = FALSE,
+                                        ncells    = 67420)
```

Comment:
Can we make this more flexible, i.e. not hardcode cell numbers?

Reply:
read.LPJ_input is from lpjclass (this is rather outdated). ncells can just have the argument 59199 and 67420. It is they way this function was written.

I will not touch this function as at some point when somebody has time we will switch to LPJmL Kit here.